osu.Framework/Graphics/Containers/HexagonalContainer.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.OpenGL.Buffers;
+using osu.Framework.Graphics.OpenGL;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Shaders;
+using osuTK.Graphics.ES30;
+using osuTK.Graphics;
+using osuTK;
+
+namespace osu.Framework.Graphics.Containers
+{
+    public class HexagonalContainer : HexagonalContainer<Drawable> { }
+
+    public class HexagonalContainer<T> : BufferedContainer<T>
+        where T : Drawable
+    {
+        private static readonly float sin_pi_over_3 = (float)Math.Sin(Math.PI / 3);
+        private static readonly float tan_pi_over_3 = (float)Math.Tan(Math.PI / 3);
+
+        public static readonly float HEXAGON_INRADIUS = sin_pi_over_3;
+
+        public IShader Shader { get; private set; }
+
+        private readonly HexagonalContainerDrawNodeSharedData sharedData;
+
+        public HexagonalContainer(RenderbufferInternalFormat[] formats = null, bool pixelSnapping = false) =>
+            sharedData = new HexagonalContainerDrawNodeSharedData(formats, pixelSnapping);
+
+        [BackgroundDependencyLoader]
+        private void load(ShaderManager shaders) => Shader = shaders.Load(VertexShaderDescriptor.TEXTURE_2, "TextureHexagon");
+
+        protected override DrawNode CreateDrawNode() => new HexagonalContainerDrawNode(this, sharedData);
+
+        public override bool ReceivePositionalInputAt(Vector2 screenSpacePos)
+        {
+            Vector2 norm = screenSpacePos - ScreenSpaceDrawQuad.TopLeft;
+            norm = new Vector2(norm.X / ScreenSpaceDrawQuad.Width, norm.Y / ScreenSpaceDrawQuad.Height); // apparently we can't divide Vector2s?
```

Comment:
it is actually possible, just not via operator overload

```suggestion
            norm = Vector2.Divide(norm, ScreenSpaceDrawQuad.Size);
```